src/cryptonote_basic/tx_extra.h

```diff
@@ -235,7 +236,7 @@ namespace cryptonote
     END_SERIALIZE()
   };
 
-  struct tx_extra_service_node_deregister
+  struct tx_extra_service_node_deregister_legacy
   {
     struct vote
```

Comment:
This struct is a nuisance for endian correction: the `votes` inside it get blob serialized, which means at the end of serialization any big endian architecture is going to have wrong validator_index values.

It can be hacked around with something like:

```C++
if (W) { for (auto &v : votes) { boost::endian::native_to_little_inplace(v); } }
FIELD(votes);
if (!W) { for (auto &v : votes) { boost::endian::little_to_native_inplace(v); } }
```
which is ugly and a pain, but since these need to be parsed historically I think some mess like that will have to go in.